### DISCUSSION TITLE
HyperSwitch SDK self-hosted static deploy

### DISCUSSION THREAD

**Discussion by filippocamagni** [juspay/hyperswitch#3650] [Q&A] [answered]
Hello!
I have a question about the hyperswitch-web repo: is it possible to deploy statically the HyperSwitch SDK and then use the static build (HyperLoader.js, app.js, etc.) in a React project without running a dedicated node server?
I am developing a React SPA that interacts with a self-hosted instance of HS. Therefore, I would like to create a “static” build that can be directly deployed on the production or development web server.
Thank you for your help.

**Comment by VedantKhairnar** [comment]
Hey @filippocamagni 
Thanks for the query.
Consider self-hosting the backend to ensure compatibility with your current version of the Hyperswitch SDK. Moreover if you self host the Hyperswitch SDK then you are essentially processing the card details so you have to be PCI Compliant.
Please let us know if you have any more queries or need any assistance.

**Reply by filippocamagni** [reply]
@VedantKhairnar, thanks for the quick reply!
Yes, I already have a self-hosted instance of the Hyperwitch server that I'm using for development. As for the SDK I managed to have a self-hosted version up and running by cloning the repo (hyperswitch-web) and starting the project with `npm run start:dev` command, but I'd like to have it built statically and deployed on common webserver.
Could you explain if it's possible to do a static build that doesn't require a dedicated node server and how to do it? (Or maybe suggest where I can find some info about this process in the documentation)?

**Reply by VedantKhairnar** [reply, marked as answer]
Ok, so by your query what I understand you just want to store the assets and access them without running an instance of the hyperswitch-web right?
I think our [production ready deployment doc](https://docs.hyperswitch.io/hyperswitch-open-source/deploy-hyperswitch-on-aws/component-wise-deployment/deploy-web-client/production-ready-deployment) follows that process. 

You can run the script which is present in that documentation or follow the steps that are written over there. You can essentially run the build command you see on the package.json and simply host the `dist` folder onto S3. You can access the `HyperLoader.js` by that path itself of the S3 we suggest having a CDN layer, and follow our general integration process.

**Reply by filippocamagni** [reply]
Yes, I tried one more time following the procedure you suggested and I was able to figure out the bug that wasn't allowing me to do the static build of the SDK `dist`. Now it works. Thank you!

**Reply by VedantKhairnar** [reply]
Great thing. Glad to hear. Happy to help!
